### PR TITLE
[Backport] [2.x] Update to Gradle 8.5 (#1131)

### PR DESCRIPTION
**Pull request by reta**
(cherry picked from commit 7c0ce4cb554a05230ecf5365b903219f357997bf)

### Description
Backport of https://github.com/opensearch-project/anomaly-detection/pull/1131 to `2.x`

### Issues Resolved
Part of https://github.com/opensearch-project/OpenSearch/issues/10334

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/ad/feature/SearchFeatureDaoTests.java [hunk 1]
```diff
@@ -265,7 +257,10 @@ public void getLatestDataTime_returnExpectedToListener() {
             return null;
         }).when(client).search(eq(searchRequest), any(ActionListener.class));
 
-        when(ParseUtils.getLatestDataTime(eq(searchResponse))).thenReturn(Optional.of(epochTime));
+        InternalMax maxAgg = new InternalMax(CommonName.AGG_NAME_MAX_TIME, epochTime, DocValueFormat.RAW, emptyMap());
```
Comment (amitgalitz) [inline src/test/java/org/opensearch/ad/feature/SearchFeatureDaoTests.java:260]: just wondering why does there need to be this new code added here

Comment (reta) [reply src/test/java/org/opensearch/ad/feature/SearchFeatureDaoTests.java:260]: This is part of phasing out PowerMock , the reference is https://github.com/opensearch-project/anomaly-detection/pull/985 (which was never backported saly), thanks @amitgalitz for looking

Comment (amitgalitz) [reply src/test/java/org/opensearch/ad/feature/SearchFeatureDaoTests.java:260]: I see, thank you for making this change!

**Review by amitgalitz — APPROVED** [review]

**Comment by jackiehanyang** [comment]
Seeing [this jackson-core error](https://github.com/opensearch-project/anomaly-detection/actions/runs/7350161068/job/20749045971) which is causing 2.x branch build failure. Should we force jackson-core version to be 2.16.1?
Reference: https://github.com/opensearch-project/flow-framework/blob/main/build.gradle#L172

**Comment by reta** [comment]
> Seeing [this jackson-core error](https://github.com/opensearch-project/anomaly-detection/actions/runs/7350161068/job/20749045971) which is causing 2.x branch build failure. Should we force jackson-core version to be 2.16.1?

Do you use explicit version ? Or it is transitive dependency (sorry, I didn't check but I will if you don't know).

**Comment by owaiskazi19** [comment]
> > Seeing [this jackson-core error](https://github.com/opensearch-project/anomaly-detection/actions/runs/7350161068/job/20749045971) which is causing 2.x branch build failure. Should we force jackson-core version to be 2.16.1?
> 
> Do you use explicit version ? Or it is transitive dependency (sorry, I didn't check but I will if you don't know).

Looks like it's coming from core as there's no occurrence of `2.16.1` in 2.x of AD . I tried to get the tree on `main` branch got the idea that it's a transitive dependency from core

```
+--- org.opensearch.test:framework:3.0.0-SNAPSHOT
|    +--- org.opensearch.client:opensearch-rest-client:3.0.0-SNAPSHOT (*)
|    +--- org.opensearch.client:opensearch-rest-client-sniffer:3.0.0-SNAPSHOT
|    |    +--- org.opensearch.client:opensearch-rest-client:3.0.0-SNAPSHOT (*)
|    |    +--- org.apache.httpcomponents.client5:httpclient5:5.2.1
|    |    +--- org.apache.httpcomponents.core5:httpcore5:5.2.2
|    |    +--- commons-codec:commons-codec:1.15
|    |    +--- commons-logging:commons-logging:1.2
|    |    \--- com.fasterxml.jackson.core:jackson-core:2.16.0
```

**Note**: The above is on the main branch.

**Comment by jackiehanyang** [comment]
> > Seeing [this jackson-core error](https://github.com/opensearch-project/anomaly-detection/actions/runs/7350161068/job/20749045971) which is causing 2.x branch build failure. Should we force jackson-core version to be 2.16.1?
> 
> Do you use explicit version ? Or it is transitive dependency (sorry, I didn't check but I will if you don't know).

nvm, we can take care of the jackson-core version in a separate pr since it's not causing build failure in this pr build. We just need to verify if it's a flaky test that is causing this build to fail - https://github.com/opensearch-project/anomaly-detection/actions/runs/7559131041/job/20750086870?pr=1132. After that we can go ahead merging this pr :)

**Comment by reta** [comment]
> Or you can just change the jackson-core version [here](https://github.com/opensearch-project/anomaly-detection/blob/main/build.gradle#L134-L135) to 2.16.1 to align with core jackson-core version that's defined here - https://github.com/opensearch-project/OpenSearch/blob/2.x/buildSrc/version.properties#L10-L11

So there is explicit version management, makes sense to change it here. I will update the pull request shortly, thanks @owaiskazi19 @jackiehanyang

**Comment by reta** [comment]
Checks are super unstable folks :(

**Comment by owaiskazi19** [comment]
@reta do we need to make the dependency upgrades on `main` first and then backport to `2.x`?

**Comment by reta** [comment]
> @reta do we need to make the dependency upgrades on `main` first and then backport to `2.x`? Otherwise, there will be discrepancies between versions.

Ah, I will do the PR in a sec for `main`

**Review by owaiskazi19 — APPROVED** [review]

**Review by jackiehanyang — APPROVED** [review]

**Pull request merged**